application/controllers/TimeframeController.php

```diff
@@ -27,7 +28,13 @@ public function editAction()
         $this->assertPermission('reporting/timeframes');
         $this->addTitleTab($this->translate('Edit Time Frame'));
 
+        $checkboxValue = 'localDateTime';
+        if (DateTime::createFromFormat('Y-m-d H:i:s', $this->timeframe->getStart()) === false) {
+            $checkboxValue = 'text';
+        }
+
         $values = [
+            'start-end' => $checkboxValue,
```

Comment:
I don't quite like that this is part of the controller action. This should be handled inside the form. There you can access the populated values with `getPopulatedValue()`.